File: Sub1AndroDevExpert/app/src/main/java/com/azhara/sub1/FilmAdapter.java
package com.azhara.sub1;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

public class FilmAdapter extends BaseAdapter {


    private Context context;
    private ArrayList<Film> films;

    public void setFilms(ArrayList<Film> films){
        this.films = films;
    }

    FilmAdapter(Context context){
        this.context = context;
        films = new ArrayList<>();
    }

    @Override
    public int getCount() {
        return films.size();
    }

    @Override
    public Object getItem(int position) {
        return films.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        if (convertView == null){
            convertView = LayoutInflater.from(context).inflate(R.layout.item_film, parent, false);
        }
        ViewHolder viewHolder = new ViewHolder(convertView);
        Film film = (Film) getItem(position);
        viewHolder.bind(film);
        return convertView;
    }


    private class ViewHolder{
        private ImageView imgFilm;
        private TextView judulFilm;
        private TextView sinopsisFilm;

        ViewHolder(View view){
            imgFilm = view.findViewById(R.id.pic_film);
            judulFilm = view.findViewById(R.id.tvJudul);
            sinopsisFilm = view.findViewById(R.id.tvSinopsis);
        }

        void bind(Film film){
            imgFilm.setImageResource(film.getPicFilm());
            judulFilm.setText(film.getJudulFilm());
            sinopsisFilm.setText(film.getSinopsisFilm());

        }
    }
}

File: Submission4Dicoding/app/src/main/java/com/azhara/submission4dicoding/ui/film/adapter/FilmAdapter.kt
package com.azhara.submission4dicoding.ui.film.adapter

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.azhara.submission4dicoding.R
import com.azhara.submission4dicoding.ui.film.model.FilmModel
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.film_item.view.*
import java.text.SimpleDateFormat
import kotlin.math.roundToInt

@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
class FilmAdapter : RecyclerView.Adapter<FilmAdapter.FilmViewHolder>() {

    private val listItem = ArrayList<FilmModel>()

    private var onItemClickCallBack: OnItemClickCallBack? = null

    fun setOnItemClickCallBack(onItemClickCallBack: OnItemClickCallBack) {
        this.onItemClickCallBack = onItemClickCallBack
    }

    fun setData(items: ArrayList<FilmModel>) {
        listItem.clear()
        listItem.addAll(items)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FilmAdapter.FilmViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.film_item, parent, false)
        return FilmViewHolder(view)
    }

    override fun getItemCount(): Int = listItem.size

    override fun onBindViewHolder(holder: FilmAdapter.FilmViewHolder, position: Int) {
        holder.bind(listItem[position])
    }

    inner class FilmViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @SuppressLint("SimpleDateFormat")
        fun bind(film: FilmModel) {
            // Change date format
            val date = film.release
            val changeTipe = SimpleDateFormat("yyyy-MM-dd")
            val format = changeTipe.parse(date)
            val changeFormat = SimpleDateFormat("d MMMM, yyyy")
            val releaseDate = changeFormat.format(format)

            // Get val vote average and change to ratingbar
            val getValueRating = film.rating?.toFloat()
            val round = getValueRating?.roundToInt()
            val rating = round?.toFloat()?.div(2)

            val posterUrl = "https://image.tmdb.org/t/p/w342/${film.poster}"
            with(itemView) {
                tv_title_film.text = film.title
                tv_release_film.text = releaseDate
                if (rating != null) {
                    rating_film.rating = rating
                }
                Glide.with(itemView)
                    .load(posterUrl)
                    .centerCrop()
                    .into(img_film)

                itemView.setOnClickListener {
                    onItemClickCallBack?.onItemClicked(film)
                }
            }
        }
    }

    interface OnItemClickCallBack {
        fun onItemClicked(data: FilmModel)
    }
}


